### ISSUE TITLE
invoke_device_method doesn't work in the sample called: iothub_registry_manager_method_sample

### ISSUE THREAD

**Issue by FelixAngelMartinez** [Azure/azure-iot-sdk-python#781]
# Context
- **OS and version used:** Windows 10
- **Python version:** 3.9.5
- **pip version:** 21.1.2
- **list of installed packages:** azure-core 1.14.0, azure-iot-device 2.6.0, azure-iot-hub 2.4.0
- **cloned repo:** master

## Description of the issue
When I try to execute the example called "iothub_registry_manager_method_sample" on my own environment, it raises an exception that doesn't allow me to receive the correct response.

## Code sample exhibiting the issue
iothub_registry_manager_method_sample

## Console log of the issue
Unexpected error JSON is invalid: Expecting ',' delimiter: line 1 column 48 (char 47)

**Comment by vishnureddy17** [comment]
Hi @FelixAngelMartinez, can you send what the device's response to the method request is when this occurs?

**Comment by FelixAngelMartinez** [comment]
It works on my device, but my backend shows me that error. I have to control that exception with a try and catch operation to avoid app crash.
The problem is in the library, more specifically in the "devices_operations.py", in the method "invoke_method" and in the part:
"deserialized = self._deserialize("CloudToDeviceMethodResult", response)".

**Comment by vishnureddy17** [comment]
When a device receives a method request, it is supposed to send back a response. This is what I meant by "device's response to the method request". Apologies if that wasn't clear. Could you send an example of a response payload sent by the device that causes this error?

**Comment by FelixAngelMartinez** [comment]
That's rigth. I've checked and the fault is in my code.
In my ESP32 device, more specifically in my c++ code I have the following line to manage the response:
const char *responseMessage =  R"({\"foo\":\"bar\"})";

And if I debug the Python SDK call, I receive the following:
<class 'requests.models.Response'>
{"status":200,"payload":{"foo":"bar"}}
Unexpected error JSON is invalid: Expecting ',' delimiter: line 1 column 38 (char 37)

I've tried to leave it empty, to change it several times in different ways, but I still don't know how to fix it.
If you have a quick solution I really appreciate it, if not, don't worry.
Thank you very much.
